openshift/e2e-tests-openshift.sh

```diff
@@ -0,0 +1,168 @@
+
+  header "Knative Build Pipeline Installed successfully"
+}
+
+function create_build_pipeline(){
+  resolve_resources config/ build-pipeline-resolved.yaml
+  oc apply -f build-pipeline-resolved.yaml
+}
+
+function resolve_resources(){
+  local dir=$1
+  local resolved_file_name=$2
+  local registry_prefix="$OPENSHIFT_REGISTRY/$OPENSHIFT_BUILD_PIPELINE_NAMESPACE/stable"
+  > $resolved_file_name
+  for yaml in $(find $dir -name "*.yaml" | grep -vE $IGNORES); do
+    echo "---" >> $resolved_file_name
+    #first prefix all test images with "test-", then replace all image names with proper repository and prefix images with "knative-build-pipeline"
+    sed -e 's%\(.* image: \)\(github.com\)\(.*\/\)\(test\/\)\(.*\)%\1\2 \3\4test-\5%' $yaml | \
+    sed -e 's%\(.* image: \)\(github.com\)\(.*\/\)\(.*\)%\1 '"$registry_prefix"'\:knative-build-pipeline-\4%' | \
+    # process these images separately as they're passed as arguments to other containers
+    sed -e 's%github.com/knative/build-pipeline/cmd/creds-init%'"$registry_prefix"'\:knative-build-pipeline-creds-init%g' | \
+    sed -e 's%github.com/knative/build-pipeline/cmd/git-init%'"$registry_prefix"'\:knative-build-pipeline-git-init%g' | \
+    sed -e 's%github.com/knative/build-pipeline/cmd/nop%'"$registry_prefix"'\:knative-build-pipeline-nop%g' \
+    >> $resolved_file_name
+  done
+}
+
+function enable_docker_schema2(){
+  oc set env -n default dc/docker-registry REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_ACCEPTSCHEMA2=true
+}
+
+function create_test_namespace(){
+  oc new-project $TEST_YAML_NAMESPACE
+  oc policy add-role-to-group system:image-puller system:serviceaccounts:$TEST_YAML_NAMESPACE -n $OPENSHIFT_BUILD_PIPELINE_NAMESPACE
+  oc new-project $TEST_NAMESPACE
+  oc policy add-role-to-group system:image-puller system:serviceaccounts:$TEST_NAMESPACE -n $OPENSHIFT_BUILD_PIPELINE_NAMESPACE
+}
+
+function run_go_e2e_tests(){
+  header "Running Go e2e tests"
+  go_test_e2e ./test/e2e/... --kubeconfig $KUBECONFIG || return 1
+}
+
+function run_yaml_e2e_tests() {
+  header "Running YAML e2e tests"
+  oc project $TEST_YAML_NAMESPACE
+  resolve_resources test/ tests-resolved.yaml
+  oc apply -f tests-resolved.yaml
+
+  # The rest of this function copied from test/e2e-common.sh#run_yaml_tests()
+  # The only change is "kubectl get builds" -> "oc get builds.build.knative.dev"
+  oc get project
+  # Wait for tests to finish.
+  echo ">> Waiting for tests to finish"
+  local tests_finished=0
+    for i in {1..60}; do
+      sleep 10
+      local finished="$(oc get builds.build.knative.dev --output=jsonpath='{.items[*].status.conditions[*].status}')"
+      if [[ ! "$finished" == *"Unknown"* ]]; then
+        tests_finished=1
+        break
+      fi
+    done
+  if (( ! tests_finished )); then
+    echo "ERROR: tests timed out"
+    return 1
+  fi
+
+  # Check that tests passed.
+  local failed=0
+  echo ">> Checking test results"
+  for expected_status in succeeded failed; do
```

Comment:
this is not the case anymore in pipeline, we don't check `succeeded` of `failed` (see https://github.com/tektoncd/pipeline/blob/master/test/e2e-tests-yaml.sh#L34). All the yaml tests are supposed to succeed :sweat_smile: